### ISSUE TITLE
hicFindTADs cannot recompute from cooler matrix

### ISSUE THREAD

**Issue by jakevc** [deeptools/HiCExplorer#377]
I am able to run hicFindTADs on a .cool file, though i get the following error when trying to test different thresholds and deltas by passing the zscore matrix. 

```
INFO:hicmatrix.lib.h5:No h5 file. Please check parameters concerning the file type!
```

It appears that section of the tool doesn't support operation on cool matrices? 

Full commands:

Initial find: 
```
hicFindTADs -m data/cool/vok2-ricky-asia.mcool::resolutions/10000 --outPrefix data/TADs/vok2-ricky-asia --correctForMultipleTesting bonferroni --numberOfProcessors 3 >logs/hicFindTADs.vok2-ricky-asia.log 2>&1
```

Recalculate on new threshold:
```
hicFindTADs -m ../cool/vok2-ricky-asia.mcool::resolutions/10000 --TAD_sep_score_prefix  vok2-ricky-asia --outPrefix vok2-ricky-asia_bf_t0.05 --thresholdComparisons 0.05 --delta 0.01 -p 8 --correctForMultipleTesting bonferroni
```

**Comment by joachimwolff** [comment]
Hi,

try the path without the leading `..`.

Best,

Joachim

**Comment by jakevc** [comment]
```
hicFindTADs -m /home/groups/hoolock/u1/jvc/TAD-RO1/hic/gibbon/hicup/data/cool/vok2-ricky-asia.mcool::resolutions/10000 --TAD_sep_score_prefix  vok2-ricky-asia --outPrefix vok2-ricky-asia_bf_t0.05 --thresholdComparisons 0.05 --delta 0.01 -p 8 --correctForMultipleTesting bonferroni
INFO:hicexplorer.hicFindTADs:
Using existing TAD-separation score file: vok2-ricky-asia_tad_score.bm

INFO:hicmatrix.lib.h5:No h5 file. Please check parameters concerning the file type!
```

**Comment by joachimwolff** [comment]
Ok, it is something more serious :( . I will have a look at it tomorrow, I think I know what is wrong: it reloads the zscore matrix but this one is stored in the wrong file format, we need to bugfix this. In the meantime: You could change the format of the zscore matrix to h5 via `hicConvertFormat zscore.cool --inputFormat cool --outputFormat h5 -o zscore.h5` and replace the current zscore with the transformed one. This should work as a workaround.

**Comment by jakevc** [comment]
Ohh interesting... The initial findTADs on my .cool produced a file ending with _zscore_matrix.h5, not _zscore_matrix.cool, so properly naming the zscore matrix suffix based on that of the input file would be a good thing also.

**Comment by joachimwolff** [comment]
Ok, than my guess is wrong. 

Do you have a `.` in the suffix? I remember that we once had troubles with that.

**Comment by jakevc** [comment]
Well it appears you guess was right. Like you said I was able to work around it by doing:

```
hicConvertFormat -m vok2-ricky-asia_zscore_matrix.h5 --inputFormat cool --outputFormat h5 -o vok2-ricky-asia_zscore_matrix.h5
```

Then:
```
hicFindTADs -m /home/groups/hoolock/u1/jvc/TAD-RO1/hic/gibbon/hicup/data/cool/vok2-ricky-asia.mcool::resolutions/10000 --TAD_sep_score_prefix  vok2-ricky-asia --outPrefix vok2-ricky-asia_bf_t0.05 --thresholdComparisons 0.05 --delta 0.01 -p 8 --correctForMultipleTesting bonferroni
INFO:hicexplorer.hicFindTADs:
Using existing TAD-separation score file: vok2-ricky-asia_tad_score.bm

INFO:hicexplorer.hicFindTADs:Computing p-values for window length: 100000

INFO:hicexplorer.hicFindTADs:Bonferroni correction. Number of boundaries for delta 0.01 and pval 0.05: 2274
```

**Comment by jakevc** [comment]
My point about the suffix is that as you see in my above example, I had to overwrite the `vok2-ricky-asia_zscore_matrix.h5` , which was not actually an .h5 because it was created from a .cool?

**Comment by joachimwolff** [comment]
No, the input data type of hicFindMatrix doesn't matter. It just means we have bug and write a cool matrix but give it the h5 file ending.

**Comment by jakevc** [comment]
Though my output prefix does contain a dot, `--outPrefix vok2-ricky-asia_bf_t0.05` it seems to have finished anyway.

**Comment by joachimwolff** [comment]
Fixed with release 3.5.
